include/libopencm3/stm32/common/spi_common_v3.h

```diff
@@ -0,0 +1,415 @@
+
+#define SPI_CFG2_MIDI_MASK						0xF
+#define SPI_CFG2_MIDI_SHIFT						4
+
+#define SPI_CFG2_MSSI_MASK						0xF
+#define SPI_CFG2_MSSI_SHIFT						0
+
+/* --- SPI_IER values ------------------------------------------------------ */
+
+#define SPI_IER_TSERFIE						(1 << 10)
+#define SPI_IER_MODFIE						(1 << 9)
+#define SPI_IER_TIFREIE						(1 << 8)
+#define SPI_IER_CRCEIE						(1 << 7)
+#define SPI_IER_OVRIE						(1 << 6)
+#define SPI_IER_UDRIE						(1 << 5)
+#define SPI_IER_TXTFIE						(1 << 4)
+#define SPI_IER_EOTIE						(1 << 3)
+#define SPI_IER_DXPIE						(1 << 2)
+#define SPI_IER_TXPIE						(1 << 1)
+#define SPI_IER_RXPIE						(1 << 0)
+
+/* --- SPI_SR values ------------------------------------------------------- */
+
+#define SPI_SR_CTSIZE_MASK						0xFFFF
+#define SPI_SR_CTSIZE_SHIFT						16
+
+#define SPI_SR_RXWNE						(1 << 15)
+
+#define SPI_SR_RXPLVL_MASK						0x3
+#define SPI_SR_RXPLVL_SHIFT						13
+
+#define SPI_SR_TXC						(1 << 12)
+#define SPI_SR_SUSP						(1 << 11)
+#define SPI_SR_TSERF						(1 << 10)
+#define SPI_SR_MODF						(1 << 9)
+#define SPI_SR_TIFRE						(1 << 8)
+#define SPI_SR_CRCE						(1 << 7)
+#define SPI_SR_OVR						(1 << 6)
+#define SPI_SR_UDR						(1 << 5)
+#define SPI_SR_TXTF						(1 << 4)
+#define SPI_SR_EOT						(1 << 3)
+#define SPI_SR_DXP						(1 << 2)
+#define SPI_SR_TXP						(1 << 1)
+#define SPI_SR_RXP						(1 << 0)
+
+/* --- SPI_IFCR values ----------------------------------------------------- */
+
+#define SPI_IFCR_SUSPC						(1 << 11)
+#define SPI_IFCR_TSERFC						(1 << 10)
+#define SPI_IFCR_MODFC						(1 << 9)
+#define SPI_IFCR_TIFREC						(1 << 8)
+#define SPI_IFCR_CRCEC						(1 << 7)
+#define SPI_IFCR_OVRC						(1 << 6)
+#define SPI_IFCR_UDRC						(1 << 5)
+#define SPI_IFCR_TXTFC						(1 << 4)
+#define SPI_IFCR_EOTC						(1 << 3)
+
+/* --- SPI_I2SCFGR values -------------------------------------------------- */
+
+#define SPI_I2SCFGR_MCKOE						(1 << 25)
+#define SPI_I2SCFGR_ODD						(1 << 24)
+
+#define SPI_I2SCFGR_I2SDIV_MASK						0xFF
+#define SPI_I2SCFGR_IS2DIV_SHIFT						16
+
+#define SPI_I2SCFGR_DATFMT						(1 << 14)
+#define SPI_I2SCFGR_WSINV						(1 << 13)
+#define SPI_I2SCFGR_FIXCH						(1 << 12)
+#define SPI_I2SCFGR_CKPOL						(1 << 11)
+#define SPI_I2SCFGR_CHLEN						(1 << 10)
+
+#define SPI_I2SCFGR_DATLEN_MASK						0x3
+#define SPI_I2SCFGR_DATLEN_SHIFT						8
+
+#define SPI_I2SCFGR_PCMSYNC						(1 << 7)
+
+#define SPI_I2SCFGR_I2SSTD_MASK						0x3
+#define SPI_I2SCFGR_I2SSTD_SHIFT						4
+
+#define SPI_I2SCFGR_I2SCFG_MASK						0x7
+#define SPI_I2SCFGR_I2SCFG_SHIFT						1
+
+#define SPI_I2SCFGR_I2SMOD						(1 << 0)
+
+#define SPI_DR8(spi_base)	MMIO8((spi_base) + 0x0c)
+#define SPI1_DR8		SPI_DR8(SPI1_BASE)
+#define SPI2_DR8		SPI_DR8(SPI2_BASE)
+#define SPI3_DR8		SPI_DR8(SPI3_BASE)
+
+/* CRCL: CRC Length */
+/****************************************************************************/
+/** @defgroup spi_crcl SPI crc length
+ * @ingroup spi_defines
+ *
+ * @{*/
+
```

Comment:
all the definitions below here looks to have been badly copied from somewhere else.

Reply:
Thanks, yes it seems so. I based my code on someone else's pull request.

I will remove the definitions for specific peripherals.

Reply:
I see definitions are shifted. I'll have a look at your review of the adc code to understand your exact requirements.

Reply:
Well I'm loosing my mind, what is your recommendation for register bits definitions? Do you have an example of code you like so I can copy it?

Reply:
It would help a lot if your comments were more actionable
What does your comment mean?
-is it better to remove these copied lines ?
-is it okay to keep them but make sure they match the fact that stm32h7 has 6 possible SPIs?
At the moment I cant tell.

Sorry for the verbosity.

Reply:
It is using a tab stop of 4px wide, so it looks misaligned and as you saw, pull request with just space adjustment take ages to get merged in, so would it be possible to adjust that straight in?

As seen in #1458 the way ahead seems to be embedding the shift in the code rather than in the register definitions.

Just caching @karlp's previous answers, not sure if I am 100% accurate.